src/ErrorBoundary.ts

```diff
@@ -18,7 +16,7 @@ const initialState: ErrorBoundaryState = {
 };
 
 export class ErrorBoundary extends Component<
-  PropsWithRef<PropsWithChildren<ErrorBoundaryProps>>,
```

Comment:
1. I'm curious whether PropsWithRef is necessary.
2. ErrorBoundaryProps need to contain children so I made this change.

Reply:
> ErrorBoundaryProps need to contain children so I made this change.

Why? (What is broken by not explicitly _requiring_ children)

Reply:
I just want to express `ErrorBoundaryProps` can contain optional children.

### `PropsWithChildren` type implematation in react
```ts
type PropsWithChildren<P = unknown> = P & { children?: ReactNode | undefined }
```

### If ErrorBoundaryProps don't contain children, Is it your intention?
in below code, prevProps contain children runtimely, but ErrorBoundaryProps won't express it as type

```ts
// ErrorBoundary.ts

componentDidUpdate(
    prevProps: ErrorBoundaryProps, // contain children runtimely, but ErrorBoundaryProps won't express it as typescriptly
    prevState: ErrorBoundaryState
  ) {
    ...
    }
  }
```

I think It's also okay not to express ErrorBoundaryProps contain chlidren exactly.
but If then, I want to ask you why don't want to express it exactly(runtimely, children can be contained in ErrorBoundaryProps) too. could you explain intention of your excluding children type?

Reply:
In TypeScript, React class components should support a `children` prop without any explicit declaration. For example, [this has no errors](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wG4AoUSWOAbziIDsATJKOAXzm1wKPRgBaZrjKVw0ePQCiUHFABCEAK4sUUAJ6duOPIWJC28wQCMVazWPJoIjAM7xmwAG5wAvHBFplIJIxgAdGj8MEjSADZIvv4AFPhOzvgAlBTkTKxQMeRwcAA8svJKqszqWpgo4eEm6ADWbrS5CQB8BdC5APTNHE3ZOXnNAMIAFsDhzEwdzb0drYrmJZpNADS9CeQp5EA). What specific error are you trying to fix?

Reply:
I know there is no error now too, but I just want to express type exactly.

Check its type error if remove PropsWithChildren in this branch like this gif
![Jul-25-2023 05-04-59](https://github.com/bvaughn/react-error-boundary/assets/61593290/a5096839-68fa-41ee-970f-5cb575de1159)

I think the lack of errors in this example link you gave me is because you used PropsWithChildren on [this line](https://github.com/bvaughn/react-error-boundary/blob/master/src/ErrorBoundary.ts#L21)

So I mean, Wouldn't it be correct that ErrorBoundaryProps contain PropsWithChildren?
If children appear like this in the console.log inside componentDidUpdate here, isn't it correct to define the type like the Pull Request I made?

![Jul-25-2023 05-12-16](https://github.com/bvaughn/react-error-boundary/assets/61593290/785aa8e0-d3b1-4bd2-a8fd-d22a8452947d)


or You think below code is right?
```ts
componentDidUpdate(
    prevProps: PropsWithChildren<ErrorBoundaryProps> // actually, I don't think this is good too, I'm just making this example to explain to you that it's a bit awkward.
    prevState: ErrorBoundaryState
  ) {
   ...
  }
```

Reply:
Sorry, a screenshot is kind of hard for me to read and understand.

Can you link me to a TypeScript playground that shows the error?

Reply:
[link what I want to explain](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzlApgOwCYqnAvnAMyghDgCJUBDAYxgFoMSyBuAKFEljgCUUaYANIlZw4AYRKQ06QSLjUqMFAFEANihAyBc5VGJQAkmgIRto4AGcAapVXAMajVrkTwEaWhgAVAJ5gUZnAAYgCuaLTA7q5SzqIACsRgFgDqwDAAFmLpwKoYqGiBvPyOmp6FfLQAchBYgQp8ShKeKAAesvGJKWnpvATa+EQk5FS0LKysrZzwMH4ocLr6AEIQYRiUUD5NSm2+-nAAvMKiGPZilDDU6QBccABGEBDqlGhsotj6N88+r8goFigwBbQZardY+G4ACgAdDD1gBzCyfNA+ADaAF0AJQHAB8cAAbhB7GxcGwJi0pvJ3BZ4ECoCDMGCtq14Ps5PVzigmW0ADy0+lrDZc7yzOAAHzgaBCqlU2IhkulGNJrCw1FU6zmBDCEXcwVsqluNAA1rxMNgIWBOjcgnqDdRDQkIEkMTcilUaihSZNoNMRdbpbb7Z0Dkc4O9oEjvnJUP9AXpgSsGRtITCofDEXAvuisftcQSiawSeMZns+QmBT4AMrpdUYB1JYMIOTuWkAfkhYagN1pQmAxggNyQ1Ek7hkFZgRpu1KgvbheGzucJGB+7l4MbbcAhcmOAMoOURW9EYsQv0oFncNzIHGw52AeJQdEoYGALAzUAREfReAPonFSCoZ7QC9DRQHwLBfC0UDxD80SPVYUAIXsUCXCVmWg2DTQQ6QMC-UR53xRcfmjAEAGkQIsddMzRYlPXJb04GLOZS1BDY6y6DJohHTxgyYxNK2rVBayDAAyEMCBtI112kO8oB+P19SNDiPBgG5FJkXYUG5OSA1Y7EfjE-0jRNLAoEkyDsGo8YvS4Bj5jjOkyzBVjUgyIzsG4uz+TBKsa1YuARMbUR9Pku1TOk2TxLtVTPFC8y5CCgNXM7ejZggAhdQMu1EosskKRsnjyyc7otKNdylgcjZvIE3z-LiiLDRdCpAXUUoYG5A8wkNNAIAAdwKA8pxnI9QnCGBIjQKL4HFBjUvEYclLkXFxXlVRwoyw0JpimTarWxLNuyqyfRLDzyp8XzWR-WyyuY07Omc9JirtORxXyxzbu6Cansu+NrsKlz0GMpU8uO66xw5Bs4BODAzgua47geJ4XlDOyIzwUkhzQak4F7NJgFsUGlC7YHePxuZDgCiHTnOS4bn0-5Ag7G5lv6GiKVVU8LC++zrtDNp-o5ia2o6R02J6eDuV+zJslyfJeSJgrOmxbFAheirxyUVhcXJ9GpxCWhoHNS1Oc8ljOixcnRAsEJ-CgA3hcVcZD3o7ILChIjYyu3jgwySxXb+AEVZ8KFbl7DAIW9ix7cd8OoWpMHDmx0a8bVj05FwB24Fj0bqDgOEAQAEWwW8kJJoJiBAWkIQZzmzYPVAYBCKA0GPSHoep+ioBCAIkf0VHU-Tt2A+hWE33TSia8d7X4CQDs8C952Y+T0lHeANLK7suAAEJ9kOZbx8dp2fYtYWoRXP2YBbKEIXN-e00CfeTwAi8rygG87wfJ8yDvw9cHt78D5dmMJMIQJ1xqoEmkdv59zZHNGQedKYwzXh8TmPY+yE30EYEwe9RDRyPkkE+aBWyXw7CgzBPw07QLcEpOBGAACqYA1hKE3I7CCeJWJoO+rxViX8WEk3YVzYmyc5A1wPNrR4KAoSqAgHCA2kFWJYgPCw1iUJLg5DyOgOAAB6DRcAfArHkM8eQ6QUB2idtgOYlhDGqPyFjJuHdPCgC7rcEI8AuqWOluo3syV-AWAUMAMAMBVA+HTqISezd4GXFnocaOmcU4TypFPX4MZSKgUif-KEuCLBL0PFouAXgjGoAAOQcwMUhXO+j-hwG6vk8xaUMhzG9mgWcGRzjt2AHCXOAlTHd2gHAasYB-AY3ohAOAAAqWwZ4Rl3HMU3OpiSSJkVfC-HwX8cndRQAU7C-04BWzmTAUajSukz3uNzUAmgTgckCVCA8OS8kWO6isXIcA7DAUCa09pblKAZ2Me4TZ68YDDNuPU9IxBepXMdjkiswy1lwBAJQYCGcG5ApadCxgaACnwEuMYw0XS3bJI5jqWZCEoCY2oHqOAM1qB5xoRmAgSgcCzJnhYmMVyDwrw3H-FuVN0h+SEn-HhycoQzy3jvKUqgeV-2rBYAAgnoSgmxqyNKQjI1hnRfZJLIkIXFZEMQHiwYeHBqrT5rkvtfR20g2g3C1aBL+zDUBQTgIo1VVqLA2sPP+c85BgKgU-n-H+WSo7z0AcnYBaAcZJw5BA0Qaco393+maPVoTBxSzUQUQgdVEpCAeutGBZQ01rUiX-A1wsfghPiWEqGXKhAz3wFEwNi9glPIBG4jAXgICJWDCo9xLwG1sohJymGerS0Y3gBkq0dUzohnvsQv+ogB5yzBDcaOc6PblldYWP+vbLA2DsA4ZqMgITxSNBiQdh5O0trbXGnAhxD12hLd-UMqgKm9ummlG9xpL0HG3uQTUI0xpkBPSE5Nrb23XvTZe22To71RofU+1eWaJoAebcBj9hx2RKBKPu+DObBAOtNlB2eKBH1zFNfq4FPUULdU5kw++h4KCNToB2OgxzPaoAAI4hGANGUM3QPl5uCoaIQb6M3kpwFhyhMhcOOjIDOuAkbIGO3IY7OuDcm5oZUHuzw1HHYByFFCB0eJ7DYFdSR0QeJbCdwHDJ444T0iusdtOmjbqz4B0XfPZdHDV0ydwGu11Z7kOmigLqshBZxg-u1E3SVMqllZGeLnUOlA0KHHREIW4iW4BZhDMpxu7LHaUAkegOEGRN5ftuPlxpRXRTijy2eTQEJgFKBACgrALQ8IbwAPK3AAFbGJgFCSw9WNApZRCHVomIdW4WJOMVgoSTh4mDIwagIQWrKMUOppwmmyCzf-aSfIxlqOyxXWCPjAZ9gIG5LN7EtJuQaIu7gbEB5zu3mxFkKx6BrsXbkNdgOSs5CzdYPbIAA)

<img width="825" alt="image" src="https://github.com/bvaughn/react-error-boundary/assets/61593290/10d4d7b6-7203-4c91-a5a2-388750b7de10">

I could met this typescript error, so I want to add children in ErrorBoundaryProps to express children existing in ErrorBoundary.props in both runtimely and typescriptly.

Reply:
I see. So the current types do not cause any user-facing errors? Only an internal one?

Reply:
Hmm, I think It will affect anyone who use `ErrorBoundaryProps` to make CustomErrorBoundary.
because after merging my Pull Request, `ErrorBoundaryProps` user don't have to add children type for CustomErrorBoundaryProps to make CustomErrorBoundary


### AS IS

```tsx
type CustomErrorBoundaryProps = PropsWithChildren<ErrorBoundaryProps & {
  somethingNew: any;
}>

const CustomErrorBoundary = ({somethingNew, ...errorBoundaryProps}) => {
  ...

  return <ErrorBoundary {...} />
} 
```

### TO BE

```tsx
type CustomErrorBoundaryProps = ErrorBoundaryProps & {
  somethingNew: any;
}

const CustomErrorBoundary = ({somethingNew, ...errorBoundaryProps}) => {
  ...

  return <ErrorBoundary {...} />
} 
```

Reply:
That's fair. I guess I haven't really anticipated anyone ever doing that (or needing to do that) but it's not like the change is harmful in some other way.